diff --git a/platforms.js b/platforms.js
index 5bb82f1..13010d1 100644
--- a/platforms.js
+++ b/platforms.js
@@ -24,6 +24,12 @@ module.exports = {
         url    : 'https://git-wip-us.apache.org/repos/asf?p=cordova-ios.git',
         version: '3.4.1'
     }, 
+    'osx' : {
+        hostos : ['darwin'],
+        parser : './src/metadata/ios_parser',
+        url    : 'https://git-wip-us.apache.org/repos/asf?p=cordova-osx.git',
+        version: 'master'
+    },
     'android' : {
         parser : './src/metadata/android_parser',
         url    : 'https://git-wip-us.apache.org/repos/asf?p=cordova-android.git',
diff --git a/src/platform.js b/src/platform.js
index 1f21f22..cf983a1 100644
--- a/src/platform.js
+++ b/src/platform.js
@@ -345,7 +345,7 @@ function call_into_create(target, projectRoot, cfg, libDir, template_dir, opts)
                 if (semver.gt(platformVersion, '3.3.0')) {
                     args.push('--cli');
                 }
-            } else if (target == 'ios') {
+            } else if (target == 'ios' || target == 'osx') {
                 var platformVersion = fs.readFileSync(path.join(libDir, 'CordovaLib', 'VERSION'), 'UTF-8').trim();
                 args.push('--arc');
                 if (semver.gt(platformVersion, '3.3.0')) {
